File: conv.c
#include <stdio.h>

#define StartMask 1 << 0
#define AMask     1 << 1
#define BMask     1 << 2
#define XMask     1 << 3
#define YMask     1 << 4
#define ZMask     1 << 5
#define UpMask    1 << 6
#define DownMask  1 << 7

int main(int argc, char* argv[]) {
    FILE* file = fopen("conv.dtm", "w");

    // skip header
    for (int i = 0; i < 0x100; i++) {
        fputc(getchar(), file);
    }

    // skip ready, go
    for (int i = 0; i < 123 * 32; i++) {
        fputc(getchar(), file);
    }

    const char* blank = "\0\0\0\0";
    char initseq[] = {StartMask, UpMask, YMask|DownMask, StartMask};
    char seq[] = {StartMask, XMask|DownMask, XMask|DownMask, XMask|DownMask, XMask|DownMask, StartMask};

    char c;
    int count = 0;
    while ((c = getchar()) != EOF) {
        fputc(c, file);
        count++;
        if (!(count % 32)) {
            char* ptr;
            for (int i = 0; i < 6; i++) {
                fputc(seq[i], file);
                for (int bit = 1; bit < 4; bit++) fputc('\0', file);
                for (int bit = 4; bit < 8; bit++) fputc('', file);

                for (int port = 2; port <= 4; port++) fputs(blank, file);
            }
        }
    }

    return 0;
}

File: conv.py
#!/usr/bin/env python2

from sys import argv
import struct

(StartMask, AMask, BMask, XMask,
    YMask, ZMask, UpMask, DownMask) = (1 << x for x in range(8))

blank = "".join(chr(x) for x in (0, 0, 0, 0, 0x80, 0x80, 0x80, 0x80))

WAIT_FRAMES = 1165 * 2


def __main__():
    if len(argv[1:]) == 2:
        infile = argv[1]
        outfile = argv[2]
    else:
        print("usage: python2 conv.py infile outfile")
        exit(1)

    with open(outfile, "w") as f:
        data = open(infile, "r").read()

        VIs, inputs, lag = struct.unpack("<3Q", data[0xD:0x25])
        data = data[:0xD] + struct.pack("<2Q", (VIs + lag) * 6, inputs * 6) + data[0x1D:]

        f.write(data[0:0x100])                           # header
        f.write(data[0x100:0x100 + WAIT_FRAMES * 0x20])  # skip to "ready, go"

        for input in (StartMask, UpMask, YMask | DownMask, StartMask):
            f.write(chr(input))
            f.write(blank[1:])
            for port in range(2, 4 + 1):
                f.write(blank)

        for frame in range(WAIT_FRAMES, (len(data) - 0x100) // 0x20 + 1):
            f.write(data[0x100 + 0x20 * frame:0x100 + 0x20 * (frame + 1)])
            for input in (StartMask, XMask | DownMask, XMask | DownMask, XMask | DownMask, XMask | DownMask, StartMask):
                f.write(chr(input))
                f.write(blank[1:])
                for port in range(2, 4 + 1):
                    f.write(blank)

if __name__ == "__main__":
    __main__()
